vigranumpy/test/test_segmentation.py

```diff
@@ -33,6 +33,49 @@ def test_labelMultiArray():
     _impl_test_labelMultiArray(numpy.uint8)
     _impl_test_labelMultiArray(numpy.uint32)
     _impl_test_labelMultiArray(numpy.float32)
+
+
+def _impl_test_applyMapping(dtype):
+    original = numpy.arange(100, dtype=dtype ).reshape(10,10)
+    mapping = { int(k) : int(k + 100) for k in original.flat[:] }
+
```

Comment:
>in case of the labelmap, you could use the promote type

That doesn't seem to work.  I don't think the problem here has to do with type widths, does it? After all, the `gaussianSmoothing()` example above is attempting to promote from float to double, which wouldn't lose any precision.

I think it has to do with the fact that `extract<>()` doesn't seem to recognize that objects of type `np.uint32` are integers.  Maybe we need to augment `converters.cxx` to explicitly support scalars of type `np.uintXX`.